SwiftCSVTests/DuplicateColumnNameHandlingTests.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  DuplicateColumnNameHandlingTests.swift
+//
+//
+//  Created by 胡逸飞 on 2024/4/27.
+//
+
+import Foundation
+import XCTest
+@testable import SwiftCSV
+
+class DuplicateColumnNameHandlingTests: XCTestCase {
+    
+    func testErrorOnDuplicateColumnNames() throws {
+        let csvString = """
+        id,name,age,name
+        1,John,23,John Doe
+        2,Jane,25,Jane Doe
+        """
+        
+        XCTAssertThrowsError(try CSV<Named>(string: csvString)) { error in
+            XCTAssertEqual(error as? CSVParseError, CSVParseError.generic(message: "Duplicate column names found: name"))
```

Comment:
Could you try to replace the equality check with a switch-case for this error? 

```suggestion
            switch error as? CSVParseError {
            case .generic(message: let message): XCTAssertEqual("Duplicate column names found: name", message)
            default: XCTFail("Expected CSVParseError.generic")
            }
```

If that works, please revert the `Equatable` conformance of the error afterwards.

Motivation:  there's just this one place in tests where equatability is used, and the conformance affects public SwiftCSV API. We can't undo that easily after we ship it, and there's no long-term motivation to make the errors equatable in apps. switch-case is the best bet there, too.